ext/date/php_date.c

```diff
@@ -4132,15 +4131,30 @@ PHP_METHOD(DatePeriod, __construct)
 	dpobj->current = NULL;
 
 	if (isostr) {
+		zend_replace_error_handling(EH_THROW, zend_ce_value_error, &error_handling);
 		date_period_initialize(&(dpobj->start), &(dpobj->end), &(dpobj->interval), &recurrences, isostr, isostr_len);
+		zend_restore_error_handling(&error_handling);
+		if (EG(exception)) {
+			RETURN_THROWS();
+		}
+
 		if (dpobj->start == NULL) {
-			php_error_docref(NULL, E_WARNING, "The ISO interval '%s' did not contain a start date.", isostr);
+			zend_string *func = get_active_function_or_method_name();
+			zend_value_error("%s(): Argument #1 must contain a start date, \"%s\" given", ZSTR_VAL(func), isostr);
```

Comment:
Weird that these aren't tested. Given how overloaded this signature is, I would prefer to keep the error message closer to what it was (i.e. `ISO interval must contain a start date`) rather than mentioning the argument number.

Reply:
I'ok fine with the change, and I'll add a small test coverage for this.

Reply:
I've just pushed it. And I've just realized that a few usage of `either` in our messages are not entirely appropriate, because this construct shouldn't be used with more than 2 options (at least in a formal text). So I slightly modified the TypeError message, and I want to review other usages later, when priority tasks calm down :)